odl/util/utility.py

```diff
@@ -223,31 +223,55 @@ def __new__(cls, name, this_bases, d):
     return metaclass('temporary_class', None, {})
 
 
+def cache_arguments(function):
+    """Decorate function to cache the result with given arguments.
+
+    This is equivalent to `functools.lru_cache` with Python 3, and currently
+    does nothing with Python 2 but this may change at some later point.
```

Comment:
WTF?? Okay, I guess this is still better than depending on some backport package for Python 2. The "later point" at which this changes will be the point when we drop Python 2 compatibility.

In the `future` library, ~~this seems to be available anyway: check [here](http://python-future.org/standard_library_imports.html?highlight=lru_cache)~~ OK, not without an extra `pip install`.